src/SchemaParser.php

```diff
@@ -25,95 +30,184 @@
+
         return $schemaBuilder->build();
     }
+
+    protected function createSchemaBuilderFromMixed(): MixedSchemaBuilder
+    {
+        return new MixedSchemaBuilder();
+    }
+
+    protected function createSchemaBuilderFromScalar(ScalarVariableType $variableType): AbstractSchemaBuilder
+    {
+        switch ($variableType->getType()) {
+            case ScalarVariableType::TYPE_INTEGER:
+                $schemaBuilder = new IntegerSchemaBuilder();
+                break;
+            case ScalarVariableType::TYPE_FLOAT:
+                $schemaBuilder = new FloatSchemaBuilder();
+                break;
+            case ScalarVariableType::TYPE_BOOLEAN:
+                $schemaBuilder = new BooleanSchemaBuilder();
+                break;
+            case ScalarVariableType::TYPE_STRING:
+                $schemaBuilder = new StringSchemaBuilder();
+                break;
+            default:
+                throw new LogicException(\sprintf("Unknown scalar type '%s'", $variableType->getType()));
+        }
+
+        return $schemaBuilder;
+    }
+
+    protected function createSchemaBuilderFromArray(ArrayVariableType $variableType): AbstractSchemaBuilder
+    {
+        if ($variableType->getKeyType() === null) {
+            $schemaBuilder = new ArraySchemaBuilder();
+        } else {
+            $schemaBuilder = new HashmapSchemaBuilder();
+        }
+
+        return $schemaBuilder->withItemsSchema(
+            $this->variableTypeToVariableSchema($variableType->getItemType(), null)
+        );
+    }
+
+    protected function createSchemaBuilderFromClass(
+        string $class,
+        ?ReflectionProperty $propertyReflection
+    ): AbstractSchemaBuilder {
+        if (! \class_exists($class)) {
+            throw new LogicException(\sprintf("Class '${class}' doesn't exists"));
+        }
+
+        // TODO: maybe move this to decorator?
+        if (\is_subclass_of($class, DateTimeInterface::class)) {
+            $schemaBuilder = new StringSchemaBuilder();
+            // TODO: move this to some constant?
+            $schemaBuilder->withFormat('date-time');
+            return $this->decorateSchemaBuilderByType($schemaBuilder, $propertyReflection);
+        }
+
+        $classReflexion = new ReflectionClass($class);
+
+        $propertiesSchemas = [];
+        foreach ($classReflexion->getProperties() as $propertyReflection) {
+            // if property is not public, then skip it.
+            if (! $propertyReflection->isPublic()) {
+                continue;
+            }
+
+            $propertiesSchemas[$propertyReflection->getName()] = $this->getPropertySchema($propertyReflection);
+        }
+
+        $schemaBuilder = (new ObjectSchemaBuilder())->withPropertiesSchemas($propertiesSchemas);
+        return $this->propertySchemaDecorator->decorateObjectSchemaBuilder(
+            $schemaBuilder,
+            $classReflexion,
+            $propertyReflection
```

Comment:
Sem chces asi posilat to, co prislo jako parametr, ale protoze se jmenuje stejne jako promenna ve foreach, tak se bude ve skutecnosti posilat posledni property z `$classReflexion->getProperties()`